src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/PatchCorrectorTest.kt

```diff
@@ -0,0 +1,109 @@
+        val text1 = "Hello world!"
+        val text2 = "Hello, world!"
+        val dmp = diff_match_patch()
+        val patch = dmp.patch_make(text1, text2)
+        val patchCorrector = PatchCorrector()
+        val listOfPatch = LinkedList<diff_match_patch.Patch>()
+        listOfPatch.addAll(patchCorrector.reversePatch(patch))
+        val newText = dmp.patch_apply(listOfPatch, text2)
+        assertEquals(text1, newText[0])
+    }
+
+    @Test
+    fun minusReversePatchTest() {
+        val text2 = "Hello world!"
+        val text1 = "Hello, world!"
+        val dmp = diff_match_patch()
+        val patch = dmp.patch_make(text1, text2)
+        val patchCorrector = PatchCorrector()
+        val listOfPatch = LinkedList<diff_match_patch.Patch>()
+        listOfPatch.addAll(patchCorrector.reversePatch(patch))
+        val newText = dmp.patch_apply(listOfPatch, text2)
+        assertEquals(text1, newText[0])
+    }
+
+    @Test
+    fun deletePatchWithAdd() {
+        val text1 = "Hello world"
+        val text2 = "Hello world!"
+        val text3 = "Hello, world!"
+        val dmp = diff_match_patch()
+        val patchForDelete  = dmp.patch_make(text1, text2)
+        val patch = dmp.patch_make(text2, text3)
+        val patchCorrector = PatchCorrector()
+        val deletePatch = patchCorrector.deletePatch(patchForDelete, patch, text2)
+        val res = dmp.patch_apply(deletePatch, text3)
+        assertEquals("Hello, world", res[0])
+    }
+
+    @Test
+    fun deletePatchWithDelete() {
+        val text1 = "Many many many words"
+        val text2 = "Many many words"
+        val text3 = "Many ! many words"
+        val dmp = diff_match_patch()
+        val patchForDelete  = dmp.patch_make(text1, text2)
+        val patch = dmp.patch_make(text2, text3)
+        val patchCorrector = PatchCorrector()
+        val deletePatch = patchCorrector.deletePatch(patchForDelete, patch, text2)
+        val res = dmp.patch_apply(deletePatch, text3)
+        assertEquals("Many ! many many words", res[0])
+    }
+
+    @Test
+    fun deletePatchFromLongList() {
+        val text1 = "Hello"
+        val text2 = "Hello world"
+        val text3 = "Hello beautiful world"
+        val text4 = "Hello, beautiful world life"
+        val text5 = "Hello, beautiful world life! Bye!"
+        val dmp = diff_match_patch()
+        val patchForDelete = dmp.patch_make(text1, text2)
+        val patch = dmp.patch_make(text2, text3)
+        patch.addAll(dmp.patch_make(text3, text4))
+        patch.addAll(dmp.patch_make(text4, text5))
+        val patchCorrector = PatchCorrector()
+        val deletePatch = patchCorrector.deletePatch(patchForDelete, patch, text2)
+        val res = dmp.patch_apply(deletePatch, text5)
+        assertEquals("Hello, beautiful life! Bye!", res[0])
+    }
+
+    @Test
+    fun bigFileChangeWorld() {
+        val text1 = "Alice was beginning to get very tired of sitting by her sister on the bank, and of having nothing " +
```

Comment:
Kotlin supports [multi-line string literals](https://kotlinlang.org/docs/reference/basic-types.html). You don't have to deal with concatenation anymore.

Reply:
ok